src/Sulu/Bundle/ContactBundle/Resources/public/js/components/documents-tab/main.js

```diff
@@ -35,141 +36,146 @@ define([
-                return data;
-            }
-        },
-
         initSidebar: function(url, id) {
             this.sandbox.emit('sulu.sidebar.set-widget', url + id);
         },
 
         render: function() {
-            var data = this.data;
             this.html(this.renderTemplate(this.templates[0]));
-            this.initForm(data);
-            this.bindCustomEvents();
-        },
-
-        initForm: function(data) {
-            var formObject = this.sandbox.form.create(this.form);
-            formObject.initialized.then(function() {
-                this.setForm(data);
-            }.bind(this));
-        },
-
-        setForm: function(data) {
-            this.sandbox.form.setData(this.form, data).fail(function(error) {
-                this.sandbox.logger.error("An error occured when setting data!", error);
-            }.bind(this));
+            this.startSelectionOverlay();
+            this.initList();
         },
 
         bindCustomEvents: function() {
-            this.sandbox.on('sulu.tab.save', this.save.bind(this));
-
-            this.sandbox.on('sulu.media-selection.document-selection.data-changed', function() {
-                this.sandbox.emit('sulu.tab.dirty');
+            // checkbox clicked
+            this.sandbox.on('husky.datagrid.documents.number.selections', function(number) {
+                var postfix = number > 0 ? 'enable' : 'disable';
+                this.sandbox.emit('husky.toolbar.documents.item.' + postfix, 'deleteSelected', false);
             }, this);
 
-            this.sandbox.on('sulu.contacts.account.documents.removed', this.resetAndRemoveFromCurrent.bind(this));
-            this.sandbox.on('sulu.contacts.contact.documents.removed', this.resetAndRemoveFromCurrent.bind(this));
+            this.sandbox.on('sulu.contacts.document.removed', function(id, mediaId) {
+                this.sandbox.emit('husky.datagrid.documents.record.remove', mediaId);
+            }.bind(this));
 
-            this.sandbox.on('sulu.contacts.account.documents.added', this.resetAndAddToCurrent.bind(this));
-            this.sandbox.on('sulu.contacts.contact.documents.added', this.resetAndAddToCurrent.bind(this));
+            this.sandbox.on('sulu.media-selection-overlay.documents.record-selected', this.addItem.bind(this));
+            this.sandbox.on('sulu.media-selection-overlay.documents.record-deselected', this.removeItem.bind(this));
+        },
 
-            this.sandbox.on('sulu.media-selection-overlay.document-selection.record-selected', this.selectItem.bind(this));
-            this.sandbox.on('sulu.media-selection.document-selection.record-deselected', this.deselectItem.bind(this));
-            this.sandbox.on('husky.dropzone.media-selection-document-selection.files-added', this.addedItems.bind(this));
+        addItem: function(id, item) {
+            if (this.currentSelection.indexOf(id) === -1) {
+                this.sandbox.emit('husky.datagrid.documents.record.add', item);
```

Comment:
as mentioned above, this should be thrown when recieving an "sulu.contacts.account.documents.added" event